capi/src/layout_editor_state.rs

```diff
@@ -0,0 +1,96 @@
+//! Represents the current state of the Layout Editor in order to visualize it properly.
+use crate::{c_char, output_str};
+use livesplit_core::{layout::editor::State as LayoutEditorState, settings::Value as SettingValue};
+
+/// type
+pub type OwnedLayoutEditorState = Box<LayoutEditorState>;
+/// type
+pub type NullableOwnedLayoutEditorState = Option<OwnedLayoutEditorState>;
+
+/// drop
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_drop(this: OwnedLayoutEditorState) {
+    drop(this);
+}
+
+/// Returns the number of components in the layout.
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_components_len(this: &LayoutEditorState) -> usize {
+    this.components.len()
+}
+
+/// Returns the name of the component at the specified index.
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_components_text(
+    this: &LayoutEditorState,
+    index: usize,
+) -> *const c_char {
+    output_str(&this.components[index])
+}
+
+/// Outputs a bitfield corresponding to which buttons are active.
+///
+/// The bits are as follows:
+///
+/// * `0x04` - Can remove the current component
+/// * `0x02` - Can move the current component up
+/// * `0x01` - Can move the current component down
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_buttons(this: &LayoutEditorState) -> u8 {
+    this.buttons.can_remove as u8 * 0x4
+        | this.buttons.can_move_up as u8 * 0x2
+        | this.buttons.can_move_down as u8 * 0x1
+}
+
+/// The index of the currently selected component
```

Comment:
This one is the only one that isn't a full sentence (lack of . at the end) and doesn't have a verb ("Returns ...").